osquery/tables/system/windows/processes.cpp

```diff
@@ -147,238 +214,334 @@ Status getProcList(std::set<long>& pids) {
+  /// Get the process UID and GID from its SID
+  HANDLE tok = nullptr;
+  std::vector<char> tok_user(sizeof(TOKEN_USER), 0x0);
+
+  auto ret = OpenProcessToken(proc, TOKEN_READ, &tok);
+  long elevated = -1;
+  if (ret != 0 && tok != nullptr) {
+    unsigned long tokOwnerBuffLen;
+    ret = GetTokenInformation(tok, TokenUser, nullptr, 0, &tokOwnerBuffLen);
+    if (ret == 0 && GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
+      tok_user.resize(tokOwnerBuffLen);
+      ret = GetTokenInformation(
+          tok, TokenUser, tok_user.data(), tokOwnerBuffLen, &tokOwnerBuffLen);
+    }
+
+    /// Check if the process is using an elevated token
+    TOKEN_ELEVATION elevation;
+    DWORD cb_size = sizeof(TOKEN_ELEVATION);
+    if (GetTokenInformation(tok,
+                            TokenElevation,
+                            &elevation,
+                            sizeof(TOKEN_ELEVATION),
+                            &cb_size)) {
+      elevated = elevation.TokenIsElevated;
+    }
+  }
+  r["is_elevated_token"] = INTEGER(elevated);
+  if (ret != 0 && !tok_user.empty()) {
+    auto sid = PTOKEN_OWNER(tok_user.data())->Owner;
+    r["uid"] = BIGINT(getUidFromSid(sid));
+    r["gid"] = BIGINT(getGidFromSid(sid));
+  } else {
+    r["uid"] = BIGINT(-1);
+    r["gid"] = BIGINT(-1);
   }
+  if (tok != nullptr) {
+    CloseHandle(tok);
+    tok = nullptr;
+  }
+}
+
+void genProcessTimeInfo(HANDLE& proc, Row& r) {
+  FILETIME create_time;
+  FILETIME exit_time;
+  FILETIME kernel_time;
+  FILETIME user_time;
+  auto ret =
+      GetProcessTimes(proc, &create_time, &exit_time, &kernel_time, &user_time);
+  if (ret == FALSE) {
+    r["user_time"] = BIGINT(-1);
+    r["system_time"] = BIGINT(-1);
+    r["start_time"] = BIGINT(-1);
+    r["process_uptime"] = BIGINT(-1);
+  } else {
+    // Windows stores proc times in 100 nanosecond ticks
+    ULARGE_INTEGER utime;
+    utime.HighPart = user_time.dwHighDateTime;
+    utime.LowPart = user_time.dwLowDateTime;
+    r["user_time"] = BIGINT(utime.QuadPart / 10000);
+    utime.HighPart = kernel_time.dwHighDateTime;
+    utime.LowPart = kernel_time.dwLowDateTime;
+    r["system_time"] = BIGINT(utime.QuadPart / 10000);
+    long long proc_create_time = osquery::filetimeToUnixtime(create_time);
+    r["start_time"] = BIGINT(proc_create_time);
+
+    FILETIME curr_ft_time;
+    SYSTEMTIME curr_sys_time;
+    GetSystemTime(&curr_sys_time);
+    SystemTimeToFileTime(&curr_sys_time, &curr_ft_time);
+    r["process_uptime"] =
+        BIGINT(osquery::filetimeToUnixtime(curr_ft_time) - proc_create_time);
+  }
+}
+
+void genProcRssInfo(HANDLE& proc, Row& r) {
+  PROCESS_MEMORY_COUNTERS_EX mem_ctr;
+  auto ret = GetProcessMemoryInfo(proc,
+                                  (PROCESS_MEMORY_COUNTERS*)&mem_ctr,
+                                  sizeof(PROCESS_MEMORY_COUNTERS_EX));
+
+  r["wired_size"] =
+      ret == TRUE ? BIGINT(mem_ctr.QuotaNonPagedPoolUsage) : BIGINT(-1);
```

Comment:
`ret == TRUE` -> `ret`

Reply:
I get what you’re saying, but these aren’t booleanes, they’re preprocessor macros for ain’t values :/ I’d argue for the explicit compare tbh, either that or I can change the check to be an explicit compare to the int value, but I’ve always hated MS code that leveraged the fact that 1 == true :(